### DISCUSSION TITLE
Add Option as an attribute of OptionList for easier access

### DISCUSSION THREAD

**Discussion by nmay231** [Textualize/textual#2832] [Ideas]
A common pattern with React components is to provide any child component of a list-like component as an attribute of the parent component.

Before
```python
from textual.widgets import OptionList
from textual.widgets.option_list import Option

OptionList(
  Option("example 1"),
  Option("example 2"),
)
```

After
```python
from textual.widgets import OptionList

OptionList(
  OptionList.Option("example 1"),
  OptionList.Option("example 2"),
)
```

Pros:

- Fewer imports
- Obvious pairing between parent and child components

Cons:

- We're modeling ourselves after React :thinking: /s
- More verbose (but you can just set `Option = OptionList.Option`)
- Confusion of the `Option` widget with `Option*` messages

Of course, `OptionList` is one of a few list-like components and the others might benefit from this change as well. Additionally, there are other child components like `Separator`.

I'm willing to help implement this :)

**Comment by davep** [comment]
We'll have a think on this. If done I think it would make sense to go all-in across the board, so other widgets would deserve similar changes.

**Reply by nmay231** [reply]
Sounds good :+1:

**Reply by nmay231** [reply]
Note: I did not get an email notification from this message even though it automatically subscribed me to this thread. Please ping me if I do not respond.
